model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/solve_nonhydro.py

```diff
@@ -798,26 +798,59 @@ def _determine_local_domains(self):
         )
         self._end_vertex_halo = self._grid.end_index(vertex_domain(h_grid.Zone.HALO))
 
-    def set_timelevels(self, nnow, nnew):
-        #  Set time levels of ddt_adv fields for call to velocity_tendencies
-        if self._config.itime_scheme == TimeSteppingScheme.MOST_EFFICIENT:
-            self.ntl1 = nnow
-            self.ntl2 = nnew
-        else:
-            self.ntl1 = 0
-            self.ntl2 = 0
+    def update_time_levels_for_velocity_tendencies(
+        self,
+        diagnostic_state_nh: dycore_states.DiagnosticStateNonHydro,
+        at_first_substep: bool,
+        at_initial_substep: bool,
```

Comment:
For me `at_first_substep` and `at_initial_substep` seem to be the same. Would it make sense to call it `at_initial_timestep` or even `at_initial_modelstep` and have it indicate only whether you are at the first model timestep? 

getting the first_supstep of the first timestep would then be `at_first_substep and at_initial_timestep`

Reply:
I like yoursuggestion to replace `at_initial_substep` or `l_init` in the original Fortran code by `at_initial_timestep`. I will do it.

Reply:
Yes, thank you. That was my suspicion from your flow write up on slack that the `l_recompute` is actually `at_first_substep`. But I did not want to open another bucket and thought we might fix it in another cleaning round...